Assets/_BForBoss/_Core/Scripts/Enemies/FloatingTargetBehaviour.cs

```diff
@@ -63,15 +73,49 @@ protected override void LifeCycleFinished()
             {
                 gameObject.SetActive(false);
             }
+            ToggleKnockbackState(false);
         }
 
         private void Awake()
         {
+            _rb = GetComponent<Rigidbody>();
```

Comment:
you could lazy initialized `_rb` (i.e. if `_rb == null` {_rb == AddComponent<Rigidbody>();}) . Then you can remove all the subsequent `if (_rb == null) return` in the other methods

Reply:
Then a require component rigid body should be added then.

If it's not required, then you need null checks and a log warning.

Reply:
Yeah my thinking was that we should allow for there not to be a rigidbody, and just have knockback do nothing if thats the case. I'm okay with making it `RequireComponent` though